### DISCUSSION TITLE
Calling UNIXUTILS ShellCommand from XCL

### DISCUSSION THREAD

**Discussion by pamoroso** [Interlisp/medley#1894] [General]
Is the UNIXUTILS function `ShellCommand` supposed to be called from Common Lisp? When I call `IL:ShellCommand` from a XCL Exec I get the error:

```
Undefined car of form
IL:SHELLCOMMAND
```

but `ShellCommand` works as expected from an Interlisp Exec:

![shellcommand-error](https://github.com/user-attachments/assets/59284104-3319-443f-bbff-c84622c9a1cf)

**Comment by MattHeffron** [comment]
Because the XCL exec uses the Common Lisp readtable, which converts to uppercase by default, `IL:ShellCommand` is converted to `IL:SHELLCOMMAND` as you saw. Instead, use `IL:|ShellCommand|` (the `|` characters keep the case as typed).

There's no `IL:SHELLCOMMAND` function, but there **is** an `IL:|ShellCommand|` function. (I.e., those are two different symbols and the function is on the mixed case one.)

**Reply by pamoroso** [reply]
That works thanks.

**Comment by rmkaplan** [comment]
But maybe this is something that should be changed.  Given the case issue, why is this particular function in camel case rather than changed rather than in eithr the dot or hyphen conventions that are more generally used:  SHELL.COMMAND or SHELL-COMMAND?

**Reply by pamoroso** [reply]
Most of the other UNIXUTILS functions are in camel case too.

**Comment by masinter** [comment]
what is the history of ShellCommand?
